apprise/utils.py

```diff
@@ -232,9 +232,12 @@ def is_hostname(hostname, ipv4=True, ipv6=True):
     # - Hostnames can ony be comprised of alpha-numeric characters and the
     #   hyphen (-) character.
     # - Hostnames can not start with the hyphen (-) character.
+    # - as a workaround for https://github.com/docker/compose/issues/229 to
+    #   being able to address services in other stacks, we also allow
+    #   underscores in hostnames
     # - labels can not exceed 63 characters
     allowed = re.compile(
-        r'(?!-)[a-z0-9][a-z0-9-]{1,62}(?<!-)$',
+        r'^[a-z0-9_][a-z0-9_-]{1,62}(?<!-)$',
```

Comment:
I removed the underscore from the first character group.

I replaced the negative lookahead because:
- it either is redundant - if Python matches the whole string against the pattern, the first character group `[a-z0-9]` already defines the possible values for the first value,
- or it is not working - if you would explicitly have to exclude a leading dash like this, you would also have to explictly exclude any other non-valid character via a negative lookahead.

So I replaced it with a `^` which enforces the pattern match from the beginning of the string.